ci-operator/step-registry/fips-check/node-scan/fips-check-node-scan-commands.sh

```diff
@@ -0,0 +1,167 @@
+attempt=0
+if [ "$http_proxy" != "" ] && [ "$https_proxy" != "" ]; then
+    while true; do
+        out=$(oc --request-timeout=60s -n "$project" debug node/"$master_node_0" -- chroot /host bash -c "export http_proxy=$http_proxy;export https_proxy=$https_proxy; curl -I ifconfig.me --connect-timeout 5")
+        if [[ $out == *"HTTP"* ]]; then
+            echo "This is not a disconnected cluser"
+            break
+        fi
+        attempt=$(( attempt + 1 ))
+        if [[ $attempt -gt 3 ]]; then
+            echo "This is a disconnected cluser, skip testing"
+            oc delete ns "$project"
+            exit 0
+        fi
+        sleep 5
+    done
+else
+    while true; do
+        out=$(oc --request-timeout=60s -n "$project" debug node/"$master_node_0" -- chroot /host bash -c "curl -I ifconfig.me --connect-timeout 5")
+        if [[ $out == *"HTTP"* ]]; then
+            echo "This is not a disconnected cluser"
+            break
+        fi
+        attempt=$(( attempt + 1 ))
+        if [[ $attempt -gt 3 ]]; then
+            echo "This is a disconnected cluser, skip testing"
+            oc delete ns "$project"
+            exit 0
+        fi
+        sleep 5
+    done
+fi
+
+# create a pod
+pod="debug-pod-$RANDOM"
+oc  -n $project debug  no/${master_node_0} --dry-run=client -o yaml > /tmp/$project-$pod.yaml
+if [ $? == 0 ]; then
+    sed -e "s/^  name: .*$/  name: $pod/" -e 's/value: "900"/value: "90000"/' /tmp/$project-$pod.yaml | oc create -f -
+    echo "sleep 10s to wait debug-pod ready."
+    COUNTER=0
+    while [ $COUNTER -lt 60 ]
+    do
+        sleep 10
+        STATUS=`oc -n $project get pod $pod -o=jsonpath="{.status.phase}"`
+        if [[ $STATUS = "Running" ]]; then
+            echo "Pod $pod is Running."
+            break
+        fi
+        COUNTER=`expr $COUNTER + 10`
+        echo "waiting ${COUNTER}s"
+    done
+    if [[ "$COUNTER" == 60 ]];then
+        echo "Time out! Pod is not Running"
+        exit
+    fi
+else
+    echo "fail to get dry-run debug file"
+    exit
+fi
+
+# run node scan and check the result
+report="/tmp/fips-check-node-scan.log"
+run_command "oc extract secret/pull-secret -n openshift-config --confirm --to /tmp"; ret=$?
+if [[ $ret -ne 0 ]]; then
+    echo "failed to get the auth"
+    exit
+fi
+oc -n $project cp /tmp/.dockerconfigjson $pod:/host/tmp/node_scan_auth.json >/dev/null 2>&1
+if [ "$http_proxy" != "" ] && [ "$https_proxy" != "" ]; then
+    oc -n $project exec $pod -- chroot /host bash -x -c "export http_proxy=$http_proxy; export https_proxy=$https_proxy; podman run --authfile /tmp/node_scan_auth.json --privileged -i -v /:/myroot registry.ci.openshift.org/ci/check-payload:latest scan node --root  /myroot 2>&1> $report"
+else
+    oc -n $project exec $pod -- chroot /host bash -x -c "podman run --authfile /tmp/node_scan_auth.json --privileged -i -v /:/myroot registry.ci.openshift.org/ci/check-payload:latest scan node --root  /myroot 2>&1> $report"
+fi
```

Comment:
Need not extract pull-secret and cp, you can directly use /var/lib/kubelet/config.json on the node, pull-secret is spread to node as this file